### PR TITLE
feat(carbon): make prepareProps available externally

### PR DESCRIPTION
**Pull request by skateman**
This function can be very useful when building custom carbon-based components...

### CODE REVIEW

**Review by rvsia — APPROVED** [review]

**Review by rvsia — APPROVED** [review]

**Comment by Hyperkid123** [comment]
@rvsia can you add the `.d.ts` file?

File: packages/carbon-component-mapper/src/files/radio.js [hunk 1]
```diff
@@ -4,7 +4,7 @@ import { useFieldApi } from '@data-driven-forms/react-form-renderer';
 
 import { FormGroup, RadioButtonGroup, RadioButton } from 'carbon-components-react';
 
-import prepareProps from '../common/prepare-props';
+import prepareProps from '.prepare-props';
```
Comment (rvsia) [inline packages/carbon-component-mapper/src/files/radio.js:7]: Bad path

**Comment by rvsia** [comment]
@Hyperkid123 me? 😄

File: packages/carbon-component-mapper/src/files/radio.js (hunk 1 above)
Comment (skateman) [reply packages/carbon-component-mapper/src/files/radio.js]: Fixed

**Comment by skateman** [comment]
Any update on this? I have no idea how to write the ts thingie :confused:

**Comment by Hyperkid123** [comment]
@skateman sorry I was out. I will try to get to it today. Hopefully, I will have a small window.

**Comment by Hyperkid123** [comment]
@skateman 
I don't have access to your fork
Here is a diff
```diff
commit 1097fc23178ae56279858adfb8ad9e54f46c23a5
Author: Martin Marosi <marvusm.mmi@gmail.com>
Date:   Thu Nov 12 13:27:13 2020 +0100

    fix(carbon): add typings to prepareProps module

diff --git a/packages/carbon-component-mapper/src/files/prepare-props.d.ts b/packages/carbon-component-mapper/src/files/prepare-props.d.ts
new file mode 100644
index 00000000..425b4351
--- /dev/null
+++ b/packages/carbon-component-mapper/src/files/prepare-props.d.ts
@@ -0,0 +1,23 @@
+import React from 'react';
+import { AnyObject } from '@data-driven-forms/react-form-renderer';
+
+export function buildLabel(label: React.ReactNode, isRequired: boolean): React.ReactNode | undefined;
+
+export type PreparePropsConfig<T = AnyObject> = T & {
+  isDisabled?: boolean;
+  isReadOnly?: boolean;
+  isRequired?: boolean;
+  label: React.ReactNode;
+  description: React.ReactNode;
+}
+
+type PreparePropsMappedOutput<T> = T & {
+  disabled?: boolean;
+  labelText?: React.ReactNode;
+  readOnly?: boolean;
+  description?: React.ReactNode;
+}
+
+declare function prepareProps<T = AnyObject>({ isDisabled, isReadOnly, isRequired, label, description, ...props }: PreparePropsConfig<T>): PreparePropsMappedOutput<T>
+
+export default prepareProps;

```

**Comment by skateman** [comment]
@Hyperkid123 updated, thanks!

**Review by rvsia — APPROVED** [review]

**Pull request merged**